crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -1391,22 +1392,20 @@ impl<'db> CallableBinding<'db> {
 
             for overload_index in matching_overload_indexes {
                 let overload = &self.overloads[*overload_index];
-                let Some(parameter_index) = overload.argument_parameters[argument_index] else {
-                    // There is no parameter for this argument in this overload.
-                    break;
-                };
-                // TODO: For an unannotated `self` / `cls` parameter, the type should be
-                // `typing.Self` / `type[typing.Self]`
-                let current_parameter_type = overload.signature.parameters()[parameter_index]
-                    .annotated_type()
-                    .unwrap_or(Type::unknown());
-                if let Some(first_parameter_type) = first_parameter_type {
-                    if !first_parameter_type.is_equivalent_to(db, current_parameter_type) {
-                        participating_parameter_index = Some(parameter_index);
-                        break;
+                for parameter_index in &overload.argument_parameters[argument_index] {
+                    // TODO: For an unannotated `self` / `cls` parameter, the type should be
+                    // `typing.Self` / `type[typing.Self]`
+                    let current_parameter_type = overload.signature.parameters()[*parameter_index]
+                        .annotated_type()
+                        .unwrap_or(Type::unknown());
+                    if let Some(first_parameter_type) = first_parameter_type {
+                        if !first_parameter_type.is_equivalent_to(db, current_parameter_type) {
+                            participating_parameter_index = Some(*parameter_index);
+                            break;
+                        }
+                    } else {
+                        first_parameter_type = Some(current_parameter_type);
                     }
-                } else {
-                    first_parameter_type = Some(current_parameter_type);
```

Comment:
Similar to my other comment, it'd be useful to have a test case for this new scenario as well where one of the parameter is not being considered for the filtering process.

We can use the [existing test case](https://github.com/astral-sh/ruff/blob/87f2ff9516dccddc385b4ed136700284ca3d1979/crates/ty_python_semantic/resources/mdtest/call/overloads.md?plain=1#L782-L805) as a base to create this new test case.

Reply:
I think I added what you're suggesting; let me know if you had something else in mind.  (Note that the test currently fails, but I'd like to dig into that as a follow-on PR)

Reply:
You can try looking at what the `top_Materialized_argument_types` is which I think for the test case should be the top materialized types for `Any` and `Literal[True]` so, `vec![object, Literal[True]]` which basically gets merged back into `tuple[object, Literal[True]]`.